api/v1alpha2/kubeadmbootstrapconfig_types.go

```diff
@@ -36,6 +36,14 @@ type KubeadmConfigSpec struct {
 	// AdditionalUserDataFiles specifies extra files to be passed to user_data upon creation.
 	// +optional
 	AdditionalUserDataFiles []Files `json:"additionalUserDataFiles,omitempty"`
+	// AdditionalPreCommands specifies extra commands to be passed to user_data upon creation
```

Comment:
Not directly related to this PR, but we should likely avoid references to user_data, since that is an AWS specific term. @chuckha thoughts?

Reply:
Agreed, it should be `AdditionalFiles` I think, or perhaps simply `Files`. But yes, out of scope for this PR.

Reply:
I'd rephrase with something like 
```golang
... specifies extra commands to be run before kubeadm.
```